src/task_queue/database_tasks/contents_spacing_correction.py

```diff
@@ -0,0 +1,110 @@
+    b. Content type: whether content is an Action/Event/Testimonial etc.
+    c. Item name: the name or title of the content
+    d. Field name: the field name of the content to be corrected
+    e. Count: the number of occurrences of spacing in the content.
+2. Fix the spacing in the database.
+"""
+
+def write_to_csv(data):
+    response = HttpResponse(content_type="text/csv")
+    writer = csv.DictWriter(response, fieldnames=["Community", "Content Type", "Item Name", "Field Name", "Count"])
+    writer.writeheader()
+    for row in data:
+        writer.writerow(row)
+    return response.content
+def get_community(instance):
+    if instance and hasattr(instance, "community"):
+        return instance.community.name if instance.community else ""
+    elif instance and hasattr(instance, "primary_community"):
+        return instance.primary_community.name if instance.primary_community else ""
+    
+
+def get_model_instances(model_name, app_label):
+    model = apps.get_model(app_label=app_label, model_name=model_name)
+    filter_args = {} if model_name == "PastEvent" else {"is_deleted": False}
+    model_instances = model.objects.filter(**filter_args)
+    return model_instances
+
+
+def auto_correct_spacing(instance, field_name, field_value):
+    for pattern in PATTERNS:
+        field_value = field_value.replace(pattern, "") 
+    setattr(instance, field_name, field_value)
+    instance.save()
+
+
+
+def is_feature_enabled(instance):
+    communities = db_models.Community.objects.filter(is_deleted=False)
+    flag = db_models.FeatureFlag.objects.filter(key=FEATURE_FLAG_KEY).first()
+    if not flag or not flag.enabled():
+        return False
+    enabled_communities = flag.enabled_communities(communities)
+    if hasattr(instance, "community") and instance.community in enabled_communities:
+         return True
+    elif hasattr(instance, "primary_community") and instance.primary_community in enabled_communities:
+         return True
+    else:
+         return False
+
+def process_spacing_data(task=None):
+    try:
+        data = []
+        models = apps.get_models()
+        for model in models:
+            app_label = model._meta.app_label
+            for field in model._meta.fields:
+                if field.__class__.__name__ == "TextField" and app_label in ["database"]:
+                    model_name = model.__name__
+                    field_name = field.name
+                    model_instances = get_model_instances(model_name, app_label)
+                    for instance in model_instances:
+                         field_value = getattr(instance, field_name)
+                         if field_value:
+                            count = len(re.findall(PATTERN, field_value))
+                            if count > 0:
+                                if is_feature_enabled(instance):
+                                    auto_correct_spacing(instance, field_name, field_value)
+                                else:
```

Comment:
no need for the "else"
Have it always append the data for what is being changed, even if the feature is enabled